nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -423,8 +423,15 @@ mountFS() {
 
     # For bind mounts, busybox has a tendency to ignore options, which can be a
     # security issue (e.g. "nosuid"). Remounting the partition seems to fix the
-    # issue.
-    mount "/mnt-root$mountPoint" -o "remount,$optionsPrefixed"
+    # issue. This should only be done for bind and rbind mountpoints because other
+    # filesystems may have limitations or may not support remount.
+    local IFS=,
+    for opt in $options; do
+        if [[ "$opt" = bind || "$opt" = rbind ]]; then
+            mount "/mnt-root$mountPoint" -o "remount,$optionsPrefixed"
+            return
```

Comment:
Surely this needs to be `break`, not `return`, right? You're exiting the whole function with `return`, not just the loop.

Reply:
Fixed. Also, unset IFS as well in case we add to this function.